docs/src/pages/[platform]/connected-components/storage/storage-browser/react.mdx

```diff
@@ -447,6 +448,127 @@ The `useView()` hook takes a single argument which is the name of the view you w
 </Example>
 
 
+### Customize actions
+
+The Storage Browser uses actions to handle data transfer with backend services, e.g. S3. You can set the `actions` config to modify or replace default actions, or adding custom actions. The `actions` config accepts 2 categories of actions: `default` and `custom`. 
+
+#### Extend default actions
+
+The `default` actions are invoked by associated default UI views to handle data transfer with S3. They are `listLocationItems`, `copy`, `createFolder`, `delete`, `download`, `upload`.
+
+You can use `defaultActionConfigs` to access the default implementation of these actions. Below is an example to instrument the default upload action to count the files being uploaded.
+
+<Example>
+<ExampleCode>
+```ts
+import { createStorageBrowser, defaultActionConfigs } from '@aws-amplify/ui-react-storage/browser';
+
+const { StorageBrowser } = createStorageBrowser({
+  actions: {
+    default: {
+      upload: {
+        ...defaultActionConfigs.upload,
+        handler: ({ config, data, options }) =>
+          defaultActionConfigs.upload.handler({
+            config,
+            data,
+            options: {
+              ...options,
+              onSuccess: (...input) => {
+                options?.onSuccess?.(...input);
+                fileUploadCounter.add(1); // Your custom counter metrics.
+              },
```

Comment:
Do we need to spread the `input` here?

Reply:
Yeah, the `input` of `onSuccess` callback is positional.